docs/parameters_desc.md

```diff
@@ -0,0 +1,189 @@
+This document attempts to describe the important parameters in Macrobase that
+affect its behavior.
+
+We first describe workload-specific parameters, that _need_ to be specified for
+every workload. These parameters do not have default values, so Macrobase will
+throw an exception if these aren't specified.
+
+<table class="table">
+<tr><th>Name</th><th>Default</th><th>Meaning</th></tr>
+<tr>
+  <td><code>macrobase.query.name</code></td>
+  <td>(none)</td>
+  <td>
+  Human-readable task name, used to identify the workload.
+  </td>
+</tr>
+<tr>
+  <td><code>macrobase.loader.attributes</code></td>
+  <td>(none)</td>
+  <td>
+  List of attributes to group by. These
+must be categorical.
+  </td>
+</tr>
+  <td><code>macrobase.loader.targetLowMetrics</code></td>
+  <td>(none)</td>
+  <td>
+  List of metrics to detect outliers from.
+These metrics have <i>low</i> values, and so are labeled as low so that Macrobase's
+outlier detection algorithms can gain more separation by looking at their
+reciprocals. Must be a floating point number.
+  </td>
+</tr>
+<tr>
+  <td><code>macrobase.loader.targetHighMetrics</code></td>
+  <td>(none)</td>
+  <td>
+  List of metrics to detect outliers from (along with <code>targetLowMetrics</code>. These
+metrics have <i>high</i> values, and can be easily separated as is. Must be a floating
+point number.
+  </td>
+</tr>
+<tr>
+  <td><code>macrobase.loader.baseQuery</code></td>
+  <td>(none)</td>
+  <td>
+  /Query used to specify input data source. Can be used to restrict
+the rows or columns Macrobase needs to look at.
+  </td>
+</tr>
+</table>
+
+The following are generic parameters, used in both our streaming and batching
+workloads.
+
+<table class="table">
+<tr><th>Name</th><th>Default</th><th>Meaning</th></tr>
+<tr>
+  <td><code>macrobase.analysis.minSupport</code></td>
+  <td><code>0.001</code></td>
+  <td>
+  Only attributes (or groups of attributes) that have
+relative count greater than this value are returned.
+  </td>
+</tr>
+<tr>
+  <td><code>macrobase.analysis.minOIRatio</code></td>
+  <td><code>3.0</code></td>
+  <td>
+  Only attributes (or groups of attributes) that occur
+in the outliers <code>minInlierRatio</code> more times than in the inliers are returned.
+  </td>
+</tr>
+  <td><code>macrobase.analysis.zScore</code> and <code>macrobase.analysis.useZScore</code></td>
+  <td><code>3.0</code> and <code>false</code></td>
+  <td>
+  Specifies zScore threshold to distinguish the
+outliers from the inliers. The zScore is as defined by the detection mechanism's
+<code>getZScoreEquivalent</code> method.
+  </td>
+</tr>
+<tr>
+  <td><code>macrobase.analysis.targetPercentile</code> and <code>macrobase.analysis.usePercentile</code></td>
+  <td><code>0.99</code> and <code>true</code></td>
+  <td>
+  Specifies percentile threshold to
```

Comment:
"macrobase.analysis.usePercnetile: if true, determine outliers according to a fixed percentile of scores. Points with scores in the `targetPercentile` percentile or greater will be marked as outliers."